doc/devdocs/locks.rst

```diff
@@ -0,0 +1,128 @@
+.. currentmodule:: Base
+
+****************************************************
+Proper maintenance and care of multi-threading locks
+****************************************************
+
+The following strategies are used to ensure that the code is dead-lock free
+(generally by addressing the 4th Coffman condition: circular wait).
```

Comment:
The purpose of this document is for people who do understand what they are doing, and have commit rights, to have a reference document against which to validate changes.